LP-2276 Fix all linting issues of homepage and student_dashboard

zahid-arbisoft opened the pull request:
### Description

[LP-2276](https://philanthropyu.atlassian.net/browse/LP-2276) Fix all linting issues of homepage and student_dashboard

asadali145 commented on lms/djangoapps/student_dashboard/views.py:
```diff
@@ -37,7 +37,7 @@ def get_enrolled_past_courses(request, course_enrollments):
                 enrolled.append(course)
 
-    for card, courses in past_course_cards.items():
+    for _, courses in past_course_cards.items():
```
```suggestion
    for courses in past_course_cards.values():
```

asadali145 reviewed (approved):
A small suggestion, Rest LGTM

zahid-arbisoft commented:
> **pylint:** ❌
> **pycodestyle:** ✔️
> **iSort:** ✔️
> 
> **Please fix following pylint issues:**
> 
> ```
> ************* Module edx-platform.lms.djangoapps.homepage.custom_context_processors
> E: 4, 0: Unable to import 'edx_notifications.server.web.utils' (import-error)
> 
> -----------------------------------
> Your code has been rated at 9.44/10
> ```

@asadali145 I have updated the code. Please ignore this error, or it quality check fails. Actually, it has cached, the requirements and in that cache edx_notifications is not installed. When pushed a dummy test commit with changes in requirements, quality check passed, because it downloaded cache again. But now, I have reverted that commit. I think, old cache will be used

MunamMubashir-arbisoft reviewed (approved):
LGTM!
